### PR TITLE
feat(uve): Ask user to create page if the language doesnt exist #28958

### LINKED ISSUE #28958: UVE: Ask user to create page if the language doesn't exist
As a user, I want to be notified when selecting a language in the Universal Visual Editor that does not have a page version, so I can decide whether to create a new version in that language.

```[tasklist]
### Acceptance Criteria
- [x] The user should see a dialog box informing them that the selected language does not have a version of the page and giving them the option to create a new version in that language.
- [x] If the users select yes, we show the edit contentlet dialog to fill up
- [x] This should work exactly as it works in the old editor
- [x] We need to catch this in the URL if use try to access the page using que queryParam
```

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
- This is the PR where @oidacra did it for the old editor: https://github.com/dotCMS/core/pull/27311

### Assumptions & Initiation Needs
The user has access to the Universal Visual Editor and is trying to select a language without an existing page version.

### Quality Assurance Notes & Workarounds
This should work exactly as the old page editor.

### PR DESCRIPTION
**Pull request by zJaaal**
## Proposed changes

- Fetch languages for the current page and check whether or not it has a translation. If it has one we let the user pass, if not we prompt to know if the user wants to create a page or not.

### Screenshot

https://github.com/dotCMS/core/assets/63567962/7d20eb2d-e747-4ebc-977e-1fb765d704ed

This PR fixes: #28958

### CODE REVIEW

File: core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/dot-ema-dialog.component.spec.ts [hunk 1]
```diff
@@ -179,6 +180,26 @@ describe('DotEmaDialogComponent', () => {
...
             });
         });
+
+        it('should dispatch onHide when p-dialog hide', () => {
+            const actionSpy = jest.spyOn(component.action, 'emit');
+
+            component.addContentlet(PAYLOAD_MOCK); // This is to make the dialog open
+            spectator.detectChanges();
+
+            const pDialog = spectator.debugElement.query(By.css('p-dialog'));
```
Comment (nicobytes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/dot-ema-dialog.component.spec.ts:190]: @zJaaal I suggest using the spectator way to do this, for example:

```ts
spectator.triggerEventHandler(Dialog, 'onHide', {})
```

File: core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts [hunk 2]
```diff
@@ -124,31 +131,64 @@ describe('DotEmaShellComponent', () => {
...
-                            viewAs: {
-                                language: {
-                                    id: 1,
-                                    language: 'English',
-                                    countryCode: 'US',
-                                    languageCode: 'EN',
-                                    country: 'United States'
+                    get({ language_id }) {
+                        return {
+                            1: of({
```
Comment (nicobytes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts:136]: @zJaaal Should this part be in another file with mocks?

File: core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts [hunk 3]
```diff
@@ -643,6 +684,246 @@ describe('DotEmaShellComponent', () => {
...
+                spectator.detectChanges();
+
+                tick(1000);
+                spectator.detectChanges();
+
+                const confirmDialog = spectator.query(byTestId('confirm-dialog'));
+
+                confirmDialog
+                    .querySelector('.p-confirm-dialog-reject')
+                    .dispatchEvent(new MouseEvent('click'));
```
Comment (nicobytes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts:743]: @zJaaal I suggest using `createMouseEvent('click');` by the spectator to create mock events.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts [hunk 4]
```diff
@@ -643,6 +684,246 @@ describe('DotEmaShellComponent', () => {
...
+                });
+
+                spectator.detectChanges();
+
+                tick(1000);
+                spectator.detectChanges();
+
+                const confirmDialog = spectator.query(byTestId('confirm-dialog'));
+
+                const dialog = spectator.debugElement.query(By.css('[data-testId="ema-dialog"]'));
```
Comment (nicobytes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts:809]: @zJaaal I suggest to use directly spectator way `spectator.query(byTestId('ema-dialog'));`.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts [hunk 5]
```diff
@@ -78,6 +79,9 @@ import { NavigationBarItem } from '../shared/models';
 export class DotEmaShellComponent implements OnInit, OnDestroy {
     @ViewChild('dialog') dialog!: DotEmaDialogComponent;
     @ViewChild('pageTools') pageTools!: DotPageToolsSeoComponent;
+
+    private readonly didTranslate = signal(false);
```
Comment (nicobytes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts:83]: @zJaaal According to recents conversations in Slack, all signals should start with `$`, like this `$didTranslate`.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts [hunk 6]
```diff
@@ -174,6 +178,9 @@ export class DotEmaShellComponent implements OnInit, OnDestroy {
     private readonly activatedRoute = inject(ActivatedRoute);
     private readonly router = inject(Router);
     private readonly siteService = inject(SiteService);
+    private readonly dotMessageService = inject(DotMessageService);
```
Comment (nicobytes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts:181]: @zJaaal According to recents conversations in Slack, all private attr should start with `#`, like this `#dotMessageService.`

File: core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts [hunk 7]
```diff
@@ -643,6 +684,258 @@ describe('DotEmaShellComponent', () => {
             });
         });
 
+        describe('language checking', () => {
+            it('should not trigger the confirmation service if the page is translated to the current language', () => {
+                const confirmationServiceSpy = jest.spyOn(confirmationService, 'confirm');
```
Comment (nicobytes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts:689]: @zJaaal  I suggest avoiding to create spies manually per each test, instead if you send in `componentProviders` the service like this `mockProvider(ConfirmationService)` automatically all methods can be spied therefore avoiding to create a manual spy.

Comment (zJaaal) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts:689]: This didn't work. I just need to do that for that method so I just do it once in the `beforeEach`

**Review by nicobytes — APPROVED** [review]
LGTM

**Review by valentinogiardino — APPROVED** [review]

**Pull request merged**